### ISSUE TITLE
[2.7] Can not install bbb-webhooks

### ISSUE THREAD

**Issue by wolbernd** [bigbluebutton/bigbluebutton#18734]
**Describe the bug**

Installing the package bbb-webhooks fails with an error `npm ERR! 'unsafe-perm' is not a valid npm option`

**Installation type**
BBB installed using https://raw.githubusercontent.com/bigbluebutton/bbb-install/v2.7.x-release/bbb-install.sh 
bbb-webhooks installed manually using apt. 

**Console output**
Full output of `apt install bbb-webhooks`:
```
# apt install bbb-webhooks
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following NEW packages will be installed:
  bbb-webhooks
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 0 B/5361 kB of archives.
After this operation, 19.1 MB of additional disk space will be used.
Selecting previously unselected package bbb-webhooks.
(Reading database ... 222379 files and directories currently installed.)
Preparing to unpack .../bbb-webhooks_1%3a2.7-4_amd64.deb ...
Unpacking bbb-webhooks (1:2.7-4) ...
Setting up bbb-webhooks (1:2.7-4) ...
npm ERR! `unsafe-perm` is not a valid npm option

npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-09-07T06_03_49_617Z-debug-0.log
dpkg: error processing package bbb-webhooks (--configure):
 installed bbb-webhooks package post-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 bbb-webhooks
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Additional context**
This error seems to have occured and already been fixed in June in #18142  by @antobinary . Maybe a regression?

**Comment by wolbernd** [comment]
Workaround:
* Install bbb-webhooks -> get an Error
* Edit `/var/lib/dpkg/info/bbb-webhooks.postinst`
* Remove the lines 
```
    cd /usr/local/bigbluebutton/bbb-webhooks
    mkdir -p node_modules

    npm config set unsafe-perm true
    npm rebuild || true

```
* Save file and install bbb-webhooks again

**Comment by antobinary** [comment]
Thank you for reporting this!
I just recreated this package on 2.7 with the fix you linked to. For now the reprocessed package is in the interim `focal-27-dev` repository https://ubuntu.bigbluebutton.org/focal-27-dev/pool/main/b/bbb-webhooks/bbb-webhooks_2.7-5_amd64.deb
and it will be copied over to the `focal-270` repository when we release BBB 2.7.1 (likely in a week or so)

**Comment by wolbernd** [comment]
@antobinary Thanks for the fast fix. The linked deb-file works perfectly for me. I'll update and close this issue after checking with 2.7.1 after it is released.

**Comment by himpierre** [comment]
> Thank you for reporting this! I just recreated this package on 2.7 with the fix you linked to. For now the reprocessed package is in the interim `focal-27-dev` repository https://ubuntu.bigbluebutton.org/focal-27-dev/pool/main/b/bbb-webhooks/bbb-webhooks_2.7-5_amd64.deb and it will be copied over to the `focal-270` repository when we release BBB 2.7.1 (likely in a week or so)

The link doesn't seem to be valid anymore.

**Comment by antobinary** [comment]
I have copied the file to a spot that will not change in the coming days. https://ubuntu.bigbluebutton.org/bbb-webhooks_2.7-7_amd64.deb

**Comment by himpierre** [comment]
Thanks!

Am 11. September 2023 17:16:56 MESZ schrieb Anton Georgiev ***@***.***>:
>I have copied the file to a spot that will not change in the coming days. https://ubuntu.bigbluebutton.org/bbb-webhooks_2.7-7_amd64.deb
>

**Comment by muellerlukas** [comment]
BB 2.7.1 is released now. But the fix is not mentioned in the release notes.
Is the problem fixed in the new version?

**Comment by antobinary** [comment]
The problem is fixed. There were no code changes, just a forced rebuild of the package. I just added a line about it, thanks!

> bbb-webhooks was rebuilt to include a carried forward fix about 'unsafe-perm' #18734
